packages/rfw/README.md

```diff
@@ -208,7 +328,542 @@ class _ExampleState extends State<Example> {
+    );
+```
+
+Any arguments in the `event` get merged with the arguments passed to
+the trigger.
+
+### Developing remote widget libraries
+
+Remote widget libraries are usually defined using a Remote Flutter
+Widgets text library file (`rfwtxt` extension), which is then compiled
+into a binary library file (`rfw` extension) on the server before
+being sent to the client.
+
+The format of text library files is defined in detail in the API
+documentation of the
+[`parseLibraryFile`](https://pub.dev/documentation/rfw/latest/formats/parseLibraryFile.html)
+function.
+
+Compiling a text `rfwtxt` file to the binary `rfw` format can be done
+by calling
+[`encodeLibraryBlob`](https://pub.dev/documentation/rfw/latest/formats/encodeLibraryBlob.html)
+on the results of calling `parseLibraryFile`.
+
+The example in `example/wasm` has some [elaborate remote
+widgets](https://github.com/flutter/packages/blob/main/packages/rfw/example/wasm/logic/calculator.rfwtxt),
+including some that manipulate state (`Button`).
+
+#### State
+
+The canonical example of a state-manipulating widget is a button.
+Buttons must react immediately (in milliseconds) and cannot wait for
+logic that's possibly running on a remote server (maybe many hundreds
+of milliseconds away).
+
+The aforementioned `Button` widget in the `wasm` example tracks a
+local "down" state, manipulates it in reaction to
+`onTapDown`/`onTapUp` events, and changes the shadow and margins of
+the button based on its state:
+
+<?code-excerpt "wasm/logic/calculator.rfwtxt (Button)"?>
+```rfwtxt
+widget Button { down: false } = GestureDetector(
+  onTap: args.onPressed,
+  onTapDown: set state.down = true,
+  onTapUp: set state.down = false,
+  onTapCancel: set state.down = false,
+  child: Container(
+    duration: 50,
+    margin: switch state.down {
+      false: [ 0.0, 0.0, 2.0, 2.0 ],
+      true: [ 2.0, 2.0, 0.0, 0.0 ],
+    },
+    padding: [ 12.0, 8.0 ],
+    decoration: {
+      type: "shape",
+      shape: {
+        type: "stadium",
+        side: { width: 1.0 },
+      },
+      gradient: {
+        type: "linear",
+        begin: { x: -0.5, y: -0.25 },
+        end: { x: 0.0, y: 0.5 },
+        colors: [ 0xFFFFFF99, 0xFFEEDD00 ],
+        stops: [ 0.0, 1.0 ],
+        tileMode: "mirror",
+      },
+      shadows: switch state.down {
+        false: [ { blurRadius: 4.0, spreadRadius: 0.5, offset: { x: 1.0, y: 1.0, } } ],
+        default: [],
+      },
+    },
+    child: DefaultTextStyle(
+      style: {
+        color: 0xFF000000,
+        fontSize: 32.0,
+      },
+      child: args.child,
+    ),
+  ),
+);
+```
+
+Becasue `Container` is implemented in `createCoreWidgets` using the
+`AnimatedContainer` widget, changing the fields causes the button to
+animate. The `duration: 50` argument sets the animation speed to 50ms.
+
+#### Lists
+
+Let us consider a remote widget library that is used to render data in
+this form and is registered with the `DynamicContent` under the name
+`server`:
```

Comment:
It's the data that would be registered with the `DynamicContent` under that name (or key) `server`, right? IIUC there isn't a sense in which a remote widget library gets registered with a `DynamicContent`.

Reply:
yeah i mangled the antecedants. good catch. fixed.